tests/Unit/Evolution/Systems/Cce/Test_LinearSolve.cpp

```diff
@@ -0,0 +1,477 @@
+                  get(linear_factor_modes)[0] +
+              random_angular_data * conj(random_angular_data) *
+                  conj(get(bondi_value_modes)[0]) *
+                  get(linear_factor_of_conjugate_modes)[0] -
+              random_angular_data * get(pole_integrand_modes)[0];
+          angular_view_for_regular_integrand = 0.0;
+
+          for (size_t j = 0; j < 5; ++j) {
+            angular_view_for_regular_integrand +=
+                (-static_cast<double>(j + 1) * get(bondi_value_modes)[j + 1] -
+                 get(pole_integrand_modes)[j + 1]) *
+                random_angular_data *
+                (j == 0
+                     ? 1.0
+                     : (one_minus_y[i] == 0.0 ? 0.0 : pow(one_minus_y[i], j)));
+            for (size_t k =
+                     static_cast<size_t>(std::max(0, static_cast<int>(j) - 2));
+                 k < std::min(j + 2, size_t{3}); ++k) {
+              angular_view_for_regular_integrand +=
+                  (square(random_angular_data) *
+                       (get(linear_factor_modes)[k] *
+                        get(bondi_value_modes)[(j + 1) - k]) +
+                   random_angular_data * conj(random_angular_data) *
+                       (get(linear_factor_of_conjugate_modes)[k] *
+                        conj(get(bondi_value_modes)[(j + 1) - k]))) *
+                  (j == 0 ? 1.0
+                          : (real(one_minus_y[i]) == 0.0
+                                 ? 0.0
+                                 : pow(one_minus_y[i], j)));
+            }
+          }
+        }
+      },
+      db::get<TestHelpers::RadialPolyCoefficientsFor<BondiValueTag>>(box),
+      db::get<TestHelpers::RadialPolyCoefficientsFor<
+          Tags::PoleOfIntegrand<BondiValueTag>>>(box),
+      db::get<TestHelpers::RadialPolyCoefficientsFor<
+          Tags::LinearFactor<BondiValueTag>>>(box),
+      db::get<TestHelpers::RadialPolyCoefficientsFor<
+          Tags::LinearFactorForConjugate<BondiValueTag>>>(box));
+  ComplexDataVector expected = outer_product(
+      random_angular_data,
+      radial_vector_from_power_series(
+          get(db::get<TestHelpers::RadialPolyCoefficientsFor<BondiValueTag>>(
+              box)),
+          one_minus_y));
+
+  make_boundary_data<BondiValueTag>(make_not_null(&box),
+                                    make_not_null(&expected), l_max);
+
+  db::mutate<Tags::OneMinusY>(make_not_null(&box),
+                              TestHelpers::compute_one_minus_y, l_max);
+
+  db::mutate_apply<RadialIntegrateBondi<BondiValueTag>>(make_not_null(&box));
+
+  Approx numerical_differentiation_approximation =
+      Approx::custom()
+          .epsilon(std::numeric_limits<double>::epsilon() * 1.0e6)
+          .scale(1.0);
+
+  CHECK_ITERABLE_CUSTOM_APPROX(expected,
+                               get(db::get<BondiValueTag>(box)).data(),
+                               numerical_differentiation_approximation);
+}
```

Comment:
These three test functions are quite similar. Do you think you can remove some of the code duplication by factoring out parts of the tests?

Reply:
Aside from the box creation at the top of each, and the integration call and check at the bottom, the generation of expected values needs to be tailored to the differential equation being solved for. I'll factor out the databox creation because that is the only one of those that significantly simplifies the resulting code.